qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -289,3 +290,98 @@ def default_values(self) -> List[DefaultCalValue]:
                     defaults.append(DefaultCalValue(value, param.name, tuple(), name))
 
         return defaults
+
+
+class EchoCrossResonance(BasisGateLibrary):
+    """A library for echoed cross-resonance gates.
+
+    Here, the ECR gate is defined as cr45p - X - cr45m. Furthermore, the X gate is defined
+    outside of this library, see :class:`FixedFrequencyTransmon`.
+    """
+
+    __default_values__ = {"tamp": 0.0, "amp": 0.5, "σ": 64, "risefall": 2, "duration": 1168}
+
+    def __init__(
+        self,
+        basis_gates: Optional[List[str]] = None,
+        default_values: Optional[Dict] = None,
+        target_pulses: bool = True,
+    ):
+        """Setup the library.
+
+        Args:
+            basis_gates: The basis gates to generate.
+            default_values: A dictionary to override library default parameter values.
+            target_pulses: If True (the default) then drives will be added to the target qubit
+                during the CR tones on the control qubit.
+        """
+        self._target_pulses = target_pulses
+        super().__init__(basis_gates, default_values)
+
+    @property
+    def __supported_gates__(self) -> Dict[str, int]:
+        """The supported gates of the library are two-qubit pulses for the ecr gate."""
+        return {"cr45p": 2, "cr45m": 2, "ecr": 2}
+
+    def default_values(self) -> List[DefaultCalValue]:
+        """The default values of the CR library."""
+        defaults = []
+        for name, schedule in self.items():
+            for param in schedule.parameters:
+                if "ch" not in param.name:
+                    value = self._default_values[param.name]
+                    defaults.append(DefaultCalValue(value, param.name, tuple(), name))
+
+        return defaults
+
+    def _build_schedules(self, basis_gates: Set[str]) -> Dict[str, ScheduleBlock]:
+        """Build the schedules of the CR library."""
+
+        schedules = dict()
+
+        tamp = Parameter("tamp")
+        sigma = Parameter("σ")
+        cr_amp = Parameter("amp")
+        cr_dur = Parameter("duration")
+        cr_rf = Parameter("risefall")
+        t_chan_idx = Parameter("ch1")
+        u_chan_idx = Parameter("ch0.1")
+        t_chan = pulse.DriveChannel(t_chan_idx)
+        u_chan = pulse.ControlChannel(u_chan_idx)
+
+        if "cr45p" in basis_gates:
+            with pulse.build(name="cr45p") as cr45p:
+                pulse.play(
+                    pulse.GaussianSquare(cr_dur, cr_amp, risefall_sigma_ratio=cr_rf, sigma=sigma),
+                    u_chan,
+                )
+
+                if self._target_pulses:
+                    pulse.play(
+                        pulse.GaussianSquare(cr_dur, tamp, risefall_sigma_ratio=cr_rf, sigma=sigma),
+                        t_chan,
+                    )
+
+            schedules["cr45p"] = cr45p
+
+        if "cr45m" in basis_gates:
```

Comment:
It should. But should we? What would be the advantages of a reference here?

Reply:
The reference here makes it clear that they have the same parameters? Since user cannot check parameter identity. Same as mapto. I don't think reference here doesn't reduce data size because we need to assign cr45p anyways.

Reply:
Mainly, I am interested in how useful parameter linking is. I think almost all cases of practical relevance can be covered by using references to other schedules instead of linking parameters across schedules. The only counter-example I can think of is pulse duration, where we typically make `sx` and `x` the same length. Part of the reason duration is linked across gates though is because it is not actually calibrated usually. Instead it is fixed to a chosen value. So it doesn't feel like giving up much to lose the linking -- the library could still set the default values for both `sx` and `x` to the same value to start out.